src/Microsoft.PowerShell.ConsoleGuiTools/ConsoleGui.cs

```diff
@@ -101,7 +105,71 @@ public void Start(ApplicationData applicationData)
                 columnWidthsSum--;
             }
 
-            win.Add(new Label(GetPaddedString(gridHeaders, columnWidths, offset + offset - 1)));
+            var filterLabel = new Label("Filter")
+            {
+                X = 2
+            };
+
+            // 7 here is to reserve space for the Apply button
+            var filterFieldWidth = usableWidth - filterLabel.Text.Length - 7;
+            var filterField = new TextField(".*")
+            {
+                X = Pos.Right(filterLabel) + 1,
+                Y = Pos.Top(filterLabel),
+                CanFocus = true,
+                Width = filterFieldWidth
+            };
+            filterField.Changed += FilterField_Changed;
+
+            var filterError = new Label(string.Empty)
+            {
+                X = Pos.Right(filterLabel) + 1,
+                Y = Pos.Top(filterLabel) + 1,
+                TextColor = Terminal.Gui.Attribute.Make(Color.Red, Color.Blue), // How to get window background color?
```

Comment:
It seems we could use default ColorScheme
https://github.com/migueldeicaza/gui.cs/blob/cd684a405ce38c0173602957b4a81f0129efd324/Terminal.Gui/Drivers/ConsoleDriver.cs#L163

Reply:
Nice find.  Will update!

Reply:
The Error colorscheme is red on blue which is hard to read, opened issue https://github.com/migueldeicaza/gui.cs/issues/350 to track, but using that for now.